src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -0,0 +1,353 @@
+		line.setFinancialObjectCode(scheduledAccountingLine.getFinancialObjectCode());
+		line.setFinancialSubObjectCode(scheduledAccountingLine.getFinancialSubObjectCode());
+		line.setNonFringeAccountOverride(scheduledAccountingLine.getNonFringeAccountOverride());
+		line.setNonFringeAccountOverrideNeeded(scheduledAccountingLine.getNonFringeAccountOverrideNeeded());
+		line.setOrganizationReferenceId(scheduledAccountingLine.getOrganizationReferenceId());
+		line.setPostingYear(scheduledAccountingLine.getPostingYear());
+		line.setProjectCode(scheduledAccountingLine.getProjectCode());
+		line.setReferenceNumber(scheduledAccountingLine.getReferenceNumber());
+		line.setReferenceTypeCode(scheduledAccountingLine.getReferenceTypeCode());
+		line.setSalesTaxRequired(scheduledAccountingLine.isSalesTaxRequired());
+		line.setSubAccountNumber(scheduledAccountingLine.getSubAccountNumber());
+		return line;
+	}
+	
+	private void saveRouteAndApproveDisbursementVouchers(List<DisbursementVoucherDocument> dvs, RecurringDisbursementVoucherDocument recurringDV){
+		for (DisbursementVoucherDocument dv : dvs) {
+			try {
+				dv.getDocumentHeader().setDocumentDescription(recurringDV.getDocumentHeader().getDocumentDescription());
+				getDocumentService().saveDocument(dv);
+				List<AdHocRouteRecipient> adHocRoutingRecipients = new ArrayList();
+				getDocumentService().routeDocument(dv, "Disbursement Document created from recurring DV: " + recurringDV.getDocumentHeader().getDocumentNumber(), adHocRoutingRecipients);
+				getDocumentService().blanketApproveDocument(dv, "Disbursement Document automatically approved by recurring DV", adHocRoutingRecipients);
+				getBusinessObjectService().save(recurringDV.getRecurringDisbursementVoucherDetails());
+			} catch (WorkflowException e) {
+				LOG.error("There was an errorr trying to save our route the created Disbursement Voucher documents: ", e);
+				throw new RuntimeException(e);
+			}
+		}
+	}
+	
+	@Override
+	public List<RecurrningDisbursementVoucherPDPStatus> getPdpStatuses(RecurringDisbursementVoucherDocument recurringDV) {
+	    List<RecurrningDisbursementVoucherPDPStatus> pdpStatuses = new ArrayList<RecurrningDisbursementVoucherPDPStatus>();
+	    for (RecurringDisbursementVoucherDetail detail : recurringDV.getRecurringDisbursementVoucherDetails()) {
+	        if (!StringUtils.isEmpty(detail.getDvDocumentNumber())){ 
+	            DisbursementVoucherDocument dv;
+                try {
+                    dv = (DisbursementVoucherDocument) getDocumentService().getByDocumentHeaderId(detail.getDvDocumentNumber());
+                } catch (WorkflowException e) {
+                    LOG.error("getPdpStatuses: There was a problem getting DV from the recurring DV detail: " + e);
+                    throw new RuntimeException(e);
+                }
+	            RecurrningDisbursementVoucherPDPStatus pdpStatus = new RecurrningDisbursementVoucherPDPStatus();
```

Comment:
It's probably not a huge deal, but the setup of this pdpStatus object could be extracted out into it's own method, cleaning up this getPdpStatuses method a bit.